xds/internal/client/client.go

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ * Copyright 2019 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+// Package client contains the implementation of the xds client used by xds.
+package client
+
+import "context"
+
+// ServiceUpdate contains update about the service.
+type ServiceUpdate struct{}
+
+// ClusterUpdate contains update about the cluster.
+type ClusterUpdate struct{}
+
+// EndpointUpdate contains update about the endpoint.
+type EndpointUpdate struct{}
+
+// Client is the xds client resolver and balancer use to watch xds updates.
+//
+// All the watch methods don't block.
+type Client interface {
```

Comment:
1. I'm thinking no. And if there are more than one consumer, they should each call `watch()` and provide their own callback. The client will need handle multiple callbacks with the same data.
1. Returning a channel limits the consumer to only use channel. A callback can do everything a channel can do (it can push the data to a channel). But with a callback, the consumer also has the flexibility to process the data inline.
1. The error parameter in `callback(data, error)` can be used for this purpose.